tests/core/test_flyer.py

```diff
@@ -138,22 +139,26 @@ async def dummy_arm_2(self=None, trigger=None, num=0, exposure=None):
     return (detector_1, detector_2)
 
 
+@pytest.mark.parametrize("number_of_frames", [[1, 1, 1, 1], [2, 3, 100, 3]])
```

Comment:
I'd like another test of the `trigger_info` basemodel - that it fails if `number_of_frames` is incorrect.

Reply:
There is a test like this in the same file 

```python
async def test_hardware_triggered_flyable_too_many_kickoffs(
    RE: RunEngine, detectors: tuple[StandardDetector]
):
    trigger_logic = DummyTriggerLogic()
    flyer = StandardFlyer(trigger_logic, [], name="flyer")
    trigger_info = TriggerInfo(
        number=1, trigger=DetectorTrigger.constant_gate, deadtime=2, livetime=2
    )

    def flying_plan():
        yield from bps.stage_all(*detectors, flyer)
        assert flyer._trigger_logic.state == TriggerState.stopping

        # move the flyer to the correct place, before fly scanning.
        # Prepare the flyer first to get the trigger info for the detectors
        yield from bps.prepare(flyer, 1, wait=True)

        # prepare detectors second.
        for detector in detectors:
            yield from bps.prepare(
                detector,
                trigger_info,
                wait=True,
            )
``` 

Should I make this parametrized ?

Reply:
I would make another one (which should be parametrized). This one deals with a specific error where the trigger info is fine but you kickoff too many times.

We want a quick one which tests you can't make a `TriggerInfo` with whatever bad input (you don't need to make it a plan, just trying an initialize directly).

Reply:
Oh this test already exists (I forgot I converted this to a pydantic Model).

https://github.com/bluesky/ophyd-async/blob/e7cef5bebce8f079bf44fb842bb8a26bf2554fa3/tests/core/test_flyer.py#L316

Just add one or two with bad list values (non int value in the list and negative int values in the list).